internal/pkg/service/stream/definition/sink_table_test.go

```diff
@@ -27,75 +27,68 @@ func TestTableSink_Validation(t *testing.T) {
 		},
 		SinkID: "my-sink",
 	}
-	versioned := Versioned{
-		Version: Version{
-			Number:      1,
-			Hash:        "0123456789123456",
-			ModifiedAt:  utctime.From(time.Now()),
-			Description: "foo bar",
-		},
-	}
-	softDeletable := SoftDeletable{
-		Deleted: false,
-	}
 
 	// Test cases
-	cases := testvalidation.TestCases[Sink]{
+	cases := testvalidation.TestCases[definition.Sink]{
 		{
 			Name: "empty",
 			ExpectedError: `
 - "projectId" is a required field
 - "branchId" is a required field
 - "sourceId" is a required field
 - "sinkId" is a required field
+- "created.at" is a required field
+- "created.by" is a required field
 - "version.number" is a required field
 - "version.hash" is a required field
-- "version.modifiedAt" is a required field
+- "version.at" is a required field
+- "version.by" is a required field
 - "type" is a required field
 - "name" is a required field
 `,
-			Value: Sink{},
+			Value: definition.Sink{},
 		},
 		{
 			Name:          "nil table section",
 			ExpectedError: `"table" is a required field`,
-			Value: Sink{
+			Value: definition.Sink{
 				SinkKey:       sinkKey,
-				Versioned:     versioned,
-				SoftDeletable: softDeletable,
-				Type:          SinkTypeTable,
+				Versioned:     test.Versioned(),
+				SoftDeletable: test.SoftDeletable(),
+				Type:          definition.SinkTypeTable,
 				Name:          "My Source",
 				Description:   "My Description",
 			},
 		},
 		{
 			Name: "empty table section",
 			ExpectedError: `
-- "table.keboola.tableId" is a required field
+- "table.type" is a required field
 - "table.mapping.columns" is a required field
 `,
-			Value: Sink{
+			Value: definition.Sink{
 				SinkKey:       sinkKey,
-				Versioned:     versioned,
-				SoftDeletable: softDeletable,
-				Type:          SinkTypeTable,
+				Versioned:     test.Versioned(),
+				SoftDeletable: test.SoftDeletable(),
+				Type:          definition.SinkTypeTable,
 				Name:          "My Source",
 				Description:   "My Description",
-				Table:         &TableSink{},
+				Table:         &definition.TableSink{},
 			},
 		},
 		{
 			Name:          "long name",
 			ExpectedError: `"name" must be a maximum of 40 characters in length`,
-			Value: Sink{
+			Value: definition.Sink{
 				SinkKey:       sinkKey,
-				Versioned:     versioned,
-				SoftDeletable: softDeletable,
-				Type:          SinkTypeTable,
+				Versioned:     test.Versioned(),
+				SoftDeletable: test.SoftDeletable(),
+				Type:          definition.SinkTypeTable,
 				Name:          strings.Repeat("a", 40+1),
 				Description:   "My Description",
-				Table: &TableSink{
-					Keboola: TableSinkKeboola{TableID: keboola.MustParseTableID("in.bucket.table")},
+				Table: &definition.TableSink{
+					Type:    definition.TableTypeKeboola,
```

Comment:
What if we move the `type` into `tabletype/tables.go` . Usage would be `tabletype.Keboola` Same for SinkType. Maybe it is currently overhead but everyting is under definition. I understand the intent but we have currently in both keywords `Table` which is a bit confusing.

Reply:
:eyes:

Reply:
(I don't have any quick ideas, I don't think it's a problem)